From 7c221cffd34344e8baf80b410329c6e012a2cfde Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 2 Nov 2016 15:46:18 +0200
Subject: [PATCH] Revert "MLK-11720: mtd: gpmi: change the erase threshold for
 mx6qp,mx6ul and mx7d"

This reverts commit 46ae731a9ed39ea03ec4dbc891c3da4450add80c.
---
 drivers/mtd/nand/gpmi-nand/gpmi-lib.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nand/gpmi-lib.c b/drivers/mtd/nand/gpmi-nand/gpmi-lib.c
index 27ff4ca7ca22..605ec8bb71e5 100644
--- a/drivers/mtd/nand/gpmi-nand/gpmi-lib.c
+++ b/drivers/mtd/nand/gpmi-nand/gpmi-lib.c
@@ -327,9 +327,9 @@ int bch_set_geometry(struct gpmi_nand_data *this)
 			| BF_BCH_FLASH0LAYOUT1_DATAN_SIZE(blockn_size, this),
 			r->bch_regs + HW_BCH_FLASH0LAYOUT1);
 
-	/* Set erase threshold to ecc strength for mx6ul, mx6qp and mx7 */
+	/* Set erase threshold to gf/2 for mx6ul, mx6qp and mx7 */
 	if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7(this) || GPMI_IS_MX6UL(this))
-		writel(BF_BCH_MODE_ERASE_THRESHOLD(ecc_strength),
+		writel(BF_BCH_MODE_ERASE_THRESHOLD(gf_len/2),
 			r->bch_regs + HW_BCH_MODE);
 
 	/* Set *all* chip selects to use layout 0. */
-- 
2.17.1

